docs/pages/reference/backends.mdx

```diff
@@ -172,6 +172,143 @@ teleport:
+recommended to prevent the user from opening replication connections, or from
+connecting to databases other than the ones used for Teleport, if the PostgreSQL
+cluster is shared between Teleport and other applications.
+
+To configure Teleport to use PostgreSQL:
+- Configure all Teleport Auth servers to use the PostgreSQL back-end in the
+  `storage` section of `teleport.yaml` as shown below.
+- Deploy several auth servers connected to the PostgreSQL storage back-end.
+- Deploy several proxy nodes.
+- Make sure that the proxy instances and all Teleport resource services that
+  connect directly to the the auth service have the `auth_server` configuration
+  setting populated with the address of a load balancer for auth service
+  instances.
+
+```yaml
+teleport:
+  storage:
+    type: postgresql
+
+    # conn_string is a libpq-compatible connection string (see
+    # https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING);
+    # pool_max_conns is an additional parameter that determines the maximum
+    # number of connections in the connection pool used for the cluster state
+    # database (the change feed uses an additional connection), defaulting to
+    # a value that depends on the number of available CPUs.
+    conn_string: postgresql://user_name@database-address/teleport_backend?sslmode=verify-full&pool_max_conns=20
+
+    # An audit_events_uri with a scheme of postgresql:// will use the
+    # PostgreSQL backend for audit log storage; the URI is a libpq-compatible
+    # connection string just like the cluster state conn_string, but cannot be
+    # specified as key=value pairs. It's possible to specify completely
+    # different PostgreSQL clusters for cluster state and audit log.
+    audit_events_uri:
+      - postgresql://user_name@database-address/teleport_audit?sslmode=verify-full
+```
+
+Audit log events are periodically deleted after a default retention period of
+3766 hours (one year); it's possible to select a different retention period or
+to disable the cleanup entirely, by specifying the `retention_period` or the
+`disable_cleanup` parameters in the fragment of the URI:
+
+```yaml
+storage:
+  teleport:
+    audit_events_uri:
+      - postgresql://user_name@database-address/teleport_audit?sslmode=verify-full#disable_cleanup=false&retention_period=2160h
+```
+
+### Authentication
+
+We strongly recommend using [client
+certificates](https://www.postgresql.org/docs/current/ssl-tcp.html#SSL-CLIENT-CERTIFICATES)
+to authenticate Teleport to PostgreSQL, as well as enforcing the use of TLS and
+[verifying the server
+certificate](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBQ-SSL-CERTIFICATES)
+on the client side. If the use of passwords is unavoidable, we recommend
+configuring them in [the `~/.pgpass`
+file](https://www.postgresql.org/docs/current/libpq-pgpass.html) rather than
+storing them in the configuration file.
```

Comment:
```suggestion
storing them in Teleport's configuration file.
```

Reply:
If we provide an option to keep passwords in Teleport config file but not recommending to use it, can we just remove that option and always require .pgpass or support reading them from files in Teleport config?

Reply:
That option comes from `pgx`'s config string parsing, do we really want to start mucking around with that?

Reply:
Ah right, for some reason I thought we have a separate fileconfig option in Teleport to provide these. If we don't and it's just a connection string parsing option, then it's fine.

Reply:
I considered checking that the password is blank in the resulting configuration, but the .pgpass file and the PGPASSWORD envvar are checked at parsing time too, so we can only prohibit passwords altogether.